apps/omg_eth/test/omg_eth/signature_test.exs

```diff
@@ -0,0 +1,105 @@
+# Copyright 2018 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Eth.SignatureTest do
+  @moduledoc """
+  Tests that elixir-generated signatures can be successfully verified by contract code.
```

Comment:
to be precise this should test something stronger - the elixir-generated signatures are just test fixture here - remember that no signatures would be generated in elixir in production.

This test should test that EIP-712-compliant signatures generated _somehow_ (via Elixir code as it happens) are treated the same by both Elixir signature code and contract signature code.

So the `defp verify` here should rather get a signature and check whether contract verification is successful iff Elixir verification is successful.

EDIT after seeing the PR in `plasma-contracts` - you can even consider returning the signer address from `verify` in contract and compare this to the address Elixir gives you, instead of comparing in the contract.

Possibly then the `verify` name will not be appropriate anymore.

Reply:
> remember that no signatures would be generated in elixir in production.

Looking at this comment makes me really curious the purpose of the "end-to-end" test here? (Also, how does it "end-to-end", where to where?)

If Elixir is not what we are interested in testing here, then it means the goal here is to test the contract. Then the test should be in the contract repo and probably not here (also, this was mentioned in previous contract CR).

Reply:
Changing :pr: title to hopefully less confusing (thanks @boolafish pointing it out)
I see the purpose of test is both `child chain` & `contract` verifies signatures - or depends on EIP-712 hashing.
Do they both works the same :question: When we upgrade dependency in elixir-omg, will this ability to verify signatures in both places remains unchanged :question:

Reply:
For future reference: relevant PR for contracts: https://github.com/omisego/plasma-contracts/pull/99